In [31]:
import pandas as pd
from collections import Counter
import matplotlib.pyplot as plt
import numpy as np
import statistics 

df= pd.read_csv("/home/danielgarcia/baby_shark_dooo_dooo/input/attacks.csv", encoding='latin-1')

In [35]:
df=df.rename(columns={'Species ': 'Species','Sex ': 'Sex','Fatal (Y/N)': 'Fatal'},)

In [128]:
df=df.reindex(['Species','Type','Activity','Injury','Fatal','Country','Area','Location','Sex','Age','Year'], axis='columns')
df=df.dropna()
df = df.round(decimals=0).astype(object)

In [37]:
df['Age']=df['Age'].str.extract('(\d+)', expand=False)

In [38]:
df['Year']=df['Year'].dropna()
df['Age'] = pd.to_numeric(df['Age'])
df['Year'] = pd.to_numeric(df['Year'])
df['Age'].unique
df.loc[df['Year'] ==0, 'Year'] = 2018 - df['Age']

# ¿Cuan peligrosos son los ataques de tiburones hacia los seres humanos?

In [40]:
a=df['Year'].min() ,df['Year'].max()
df.drop(df.loc[df['Type']=='Invalid'].index, inplace=True)
print (f"El primer año con ataque registrado es {int(a[0])} y el ultimo es {int(a[1])}.")

El primer año con ataque registrado es 1868 y el ultimo es 2018.


### El DataSet  nos muestra informacion recopilada entre 1868 y 2018, esto son 150 años de informacion recopilada a largo de todo el globo terraqueo.

In [157]:
df['Year'].value_counts()
df['Year'].min()
Rango_ataques=int(df['Year'].max()-df['Year'].min())
Rango_ataques
print(f'Han transcurrido {Rango_ataques} años entre el primer ataque registrado y el ultimo.')

Han transcurrido 150 años entre el primer ataque registrado y el ultimo.


#### Puesto que queremos ver si realmente los tiburones son tan peligrosos para los humanos analizaremos los datos relacionados con el genero, la edad y si el ataque fue definitivo o se quedo en un susto.                                                                         El primer dato calculado es la edad media de los participantes en el estudio, observamos que son 28 años, la mediana con 25 y la desviacion tipica es de 13.  Los valores extremos son 84 años y 5 años, la edad dividida por cuartiles nos indican grandes varianzas, mientras que el cuartil 0,5 se situa en los 25 años (coincidiendo con la mediana y siendo un poco menor que la media) , el 3 cuantil nos situa en los 36 años vemos un tremendo gap hasta el final con los 84 años. 



In [188]:
Age_Mean=df["Age"].mean()
Age_Median=df["Age"].median()
Age_std=df["Age"].std()
print(f"La edad media de los que han sufrido un ataque es de {int(Age_Mean)}, el valor mas repetido, esto es, la mediana con {int(Age_Median)} y la desviacion es {int(Age_std)}.")
print (f"La edad maxima de los atacados es de {int(df['Age'].max())} y la minima de {int(df['Age'].min())}")
a=df['Age'].quantile([0.25,0.5,0.75,1]).quantile


La edad media de los que han sufrido un ataque es de 28, el valor mas repetido, esto es, la mediana con 25 y la desviacion es 13.
La edad maxima de los atacados es de 84 y la minima de 5


### Si empezamos a aplicar filtros veremos informacion mas precisa, yo he decidido utilizar filtros demograficos para ver si hay patrones que se repiten, tambien desglose la informacion en base a si la victima era una victima mortal o no puesto que estaba interesado en ver informacion relacionada 

In [187]:
a=df.loc[(df['Fatal']=='N') & (df['Sex']=='M' )& (df['Age']),['Fatal','Sex','Age']]
b=df.loc[(df['Fatal']=='N') & (df['Sex']=='F' )& (df['Age']),['Fatal','Sex','Age']]
c=df.loc[(df['Fatal']=='Y') & (df['Sex']=='M' )& (df['Age']),['Fatal','Sex','Age']]
d=df.loc[(df['Fatal']=='Y') & (df['Sex']=='F' )& (df['Age']),['Fatal','Sex','Age']]
print (int(a.median()[0]))
print(a.std()[0])
print (f"La edad maxima registrada del superviviente masculino es de: {int(a['Age'].max())}\nLa de las mujeres supervivientes es de: {int(b['Age'].max())}\nPor otro lado la del fallecido masculino mas adulto de: {int(c['Age'].max())} y la de la mujer fallecida es de: {int(d['Age'].max())}")


print (f"La edad minima registrada del superviviente masculino es de: {int(a['Age'].min())}\nLa edad minima de las mujeres supervivientes es de: {int(b['Age'].min())}\nPor otro lado la del fallecido masculino mas joven de: {int(c['Age'].min())} y la de la mujer fallecida es de: {int(d['Age'].min())}\n\n")



print (f'La edad media de supervivientes masculinos es de: {a.mean()[0]}\nLa de las mujeres supervivientes es de : {b.mean()[0]}\nLa edad media de los fallecidos masculinos era de: {c.mean()[0]}\nLa de las mujeres fallecidas de: {d.mean()[0]}')

25
13.239341020375448
La edad maxima registrada del superviviente masculino es de: 73
La de las mujeres supervivientes es de: 70
Por otro lado la del fallecido masculino mas adulto de: 84 y la de la mujer fallecida es de: 77
La edad minima registrada del superviviente masculino es de: 5
La edad minima de las mujeres supervivientes es de: 6
Por otro lado la del fallecido masculino mas joven de: 7 y la de la mujer fallecida es de: 7


La edad media de supervivientes masculinos es de: 28.312915129151293
La de las mujeres supervivientes es de : 27.574257425742573
La edad media de los fallecidos masculinos era de: 27.295833333333334
La de las mujeres fallecidas de: 30.976190476190474


In [47]:
def len_fa(df):
    return len(df[df['Fatal'] == 'N']),len(df[df['Fatal'] == 'Y'])
print(len_fa(df))
print(f"El numero total de supervivientes es de {len_fa(df)[0]} y de fallecidos es de {len_fa(df)[1]} .")

(1566, 283)
El numero total de supervivientes es de 1566 y de fallecidos es de 283 .


In [48]:
def len_gen(df):
    return len(df[df['Sex'] == 'F']),len(df[df['Sex'] == 'M'])
len_gen(df)
print(f"EL numero de mujeres registradas es de {len_gen(df)[0]} y el de hombres es de {len_gen(df)[1]}.")

EL numero de mujeres registradas es de 246 y el de hombres es de 1607.


In [49]:
pl=df.loc[(df['Fatal']=='N') & (df['Sex']=='M' ),['Fatal','Sex']]
(np.sum(pl, axis=1).tolist)
len(pl)
#EL numero de hombres muertos es de {len(dpm)}
print (f"EL numero de hombres vivos es de {len(pl)}.")

EL numero de hombres vivos es de 1361.


In [50]:
lmu=df.loc[(df['Fatal']=='N') & (df['Sex']=='F' ),['Fatal','Sex']]
np.sum(lmu, axis=1).tolist
len(lmu)
print (f"EL numero de mujeres vivas es de {len(lmu)}")


EL numero de mujeres vivas es de 204


In [51]:
depm=df.loc[(df['Fatal']=='Y') & (df['Sex']=='M' ),['Fatal','Sex']]
(np.sum(depm, axis=1).tolist)
len(depm)
#EL numero de hombres muertos es de {len(dpm)}
print (f"EL numero de hombres muertos es de {len(depm)}")

EL numero de hombres muertos es de 241


In [52]:
depmu=df.loc[(df['Fatal']=='Y') & (df['Sex']=='F' ),['Fatal','Sex']]
np.sum(depmu, axis=1).tolist
len(depmu)
print (f"EL numero de mujeres muertas es de {len(depmu)}")


EL numero de mujeres muertas es de 42


In [53]:
Leth= (len(df[df['Fatal'] == 'Y'])*100/len(df[df['Fatal'] == 'N']))
Leth
#La letalidad de los ataques mostrados es de 18.22

18.071519795657725

In [54]:
SuPr=df.loc[(df['Fatal']=='N') & (df['Type']=='Provoked' ),['Fatal','Type']]
LePr=df.loc[(df['Fatal']=='Y') & (df['Type']=='Provoked' ),['Fatal','Type']]
SuPr.size, LePr.size, df['Type'].size
tnp=(df['Type'].size)- (SuPr.size+LePr.size)
print (f'El total de casos provocados es de {SuPr.size+LePr.size} y el de casos no provocados: {tnp}, de los cuales solo {LePr.size} han sido provocados y mortales.')
#ataques provocados y letalidad 

El total de casos provocados es de 390 y el de casos no provocados: 1464, de los cuales solo 4 han sido provocados y mortales.


In [126]:
Surfinga=df.loc[(df['Fatal']=='N') & (df['Type']=='Provoked' ) &(df['Sex']=='M'),['Fatal','Type','Sex']].shape
b=df.loc[(df['Fatal']=='N') & (df['Type']=='Provoked' ) &(df['Sex']=='F'),['Fatal','Type','Sex']].shape
c=df.loc[(df['Fatal']=='N') & (df['Type']=='Unprovoked' ) &(df['Sex']=='M'),['Fatal','Type','Sex']].shape
d=df.loc[(df['Fatal']=='N') & (df['Type']=='Unprovoked' ) &(df['Sex']=='F'),['Fatal','Type','Sex']].shape
print(f"El numero de ataques provocados por hombres sin victimas letales es de: {a[0]} y el de mujeres es de {b[0]}\nEl numero de ataques sin provocar y sin resultado mortal a hombres es de {c[0]} y el de mujeres {d[0]}")

El numero de ataques provocados por hombres sin victimas letales es de: 1 y el de mujeres es de 12
El numero de ataques sin provocar y sin resultado mortal a hombres es de 1153 y el de mujeres 185


In [55]:
df['Type'].value_counts()

Unprovoked      1624
Provoked         197
Boating           25
Sea Disaster       7
Boat               1
Name: Type, dtype: int64

In [123]:

cuartiles=np.percentile(df.Year,[25, 50, 75])

In [59]:
coty=  df[["Country","Type",'Fatal']].sort_values(by="Country")
# Relacionar tipo de bicho con los mas letales y paises mas afectados.
a=coty.loc[(coty['Fatal']=='Y') & (coty['Country']=='USA' )& (coty['Type']=='Unprovoked'),['Country','Type','Fatal']]
a.size

120

In [116]:
plot two columns pandas

SyntaxError: invalid syntax (<ipython-input-116-18ad78b50b0e>, line 1)

In [109]:
#Con los numeros vistos, la escasa cifra de victimas mortales, el tiempo utilizado en la muestra siendo superior a un siglo, podemos asegurar que es mas factible morir por el baby shark que ante el verdadero ataque de un tiburon.